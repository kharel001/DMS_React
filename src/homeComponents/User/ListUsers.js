import React, { useEffect} from 'react';
import MaterialTable from 'material-table';
import TableIcons from '../../homeComponents/TableIcons/TableIcons';
import { useSelector, useDispatch } from 'react-redux'
import * as actions from '../../store/actions/users'
import { Redirect } from 'react-router';

/**
* A List of healthpost users
*
* @param - No parameters
* @returns {ListUsers} - returns a material table of healthpost users
*/

const ListUsers = () => {

    const [state, setState] = React.useState({
        columns: [
            { title: 'First Name', field: 'firstname' },
            { title: 'Last Name', field: 'lastname' },
            { title: 'Health Post', field: 'healthPost' },
            { title: 'Address', field: 'address' },
            { title: 'Phone Number', field: 'phonenumber' },
            { title: 'Email', field: 'email' },
            { title: 'Password', field: 'password' }
        ],
        data: [
        ],
    });
    const dispatch = useDispatch();
    const userData = useSelector(({ users }) => users.users)
    const [redirectTo, setRedirectTo] = React.useState(null);

    useEffect(() => {
        dispatch(actions.getUsers());
    }, [dispatch])

    useEffect(() => {
        setState(prevState => { return { ...prevState, data: userData.healthpost } })
    }, [userData])

    const [selectedRow, setSelectedRow] = React.useState(null);

    const selectUser = (id) => {
        // console.log(id);
        setRedirectTo(<Redirect to={{
            pathname: '/app/users/profile',
            state: { id: id }
        }} />)
    }
    // console.log(userData);
    return (
        <div className="app-wrapper">
            {redirectTo}
            <div className="animated slideInUpTiny animation-duration-3">
                <MaterialTable
                    title="Users"
                    columns={state.columns}
                    data={state.data}
                    icons={TableIcons}
                    onRowClick={((evt, selectedRow) => selectUser(selectedRow.tableData.id))}
                    options={{
                        rowStyle: rowData => ({
                            backgroundColor: (selectedRow === rowData.tableData.id) ? '#EEE' : '#FFF',
                            font: 'Roboto'
                        }),
                        headerStyle: {
                            backgroundColor: '#01579b',
                            color: '#FFF'
                        }
                    }}
                    editable={{
                        onRowAdd: (newData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    console.log(newData);
                                    //   dispatch(actions.addMedicine(newData))
                                    setState((prevState) => {
                                        const data = [...prevState.data];
                                        data.push(newData);
                                        return { ...prevState, data };
                                    });
                                }, 600);
                            }),
                        onRowUpdate: (newData, oldData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    if (oldData) {
                                        console.log(newData);
                                        // dispatch(actions.updateMedicine(newData, newData._id))
                                        setState((prevState) => {
                                            const data = [...prevState.data];
                                            data[data.indexOf(oldData)] = newData;
                                            return { ...prevState, data };
                                        });
                                    }
                                }, 600);
                            }),
                        onRowDelete: (oldData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    setState((prevState) => {
                                        const data = [...prevState.data];
                                        data.splice(data.indexOf(oldData), 1);
                                        // dispatch(actions.deleteMedicine(oldData._id))
                                        return { ...prevState, data };
                                    });
                                }, 600);
                            }),
                    }}
                />
            </div>
        </div>)
}

export default ListUsers;