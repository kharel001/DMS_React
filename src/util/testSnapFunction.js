import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


export  const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

export const findByTestProps=(wrapper,val)=>{
    return wrapper.findByProps({"data-test": val})
}


  export const mockSocket = OS => {
    jest.resetModules();
    jest.doMock('../socket', () => ({
      socket:{
          emit:jest.fn(),
          on:jest.fn(),
          off:jest.fn()

      },
      emit:jest.fn(),
      on:jest.fn(),
      off:jest.fn()
    }));
  };
  
    export const mockMaterialUI = ()=>{
      jest.doMock('@material-ui/styles', () => ({
          Theme:{
            spacing:{
              unit:20
            }
          },makeStyles:jest.fn(),
          createStyles:jest.fn()
        }));
      };
    export const mockAvatar = ()=>{
      jest.resetModules();
      jest.doMock('@material-ui/core/es/Avatar/Avatar', (props) => <div >{props.children}</div>);
      };
export const mockDevice = ()=>{
    jest.resetModules();
    jest.doMock('react-device-detect', () => ({
        isIOS:false,
        isMobile:true
      }));
    };
const useHistoryFunc=jest.fn();
const listenFunc = jest.fn();
useHistoryFunc.mockReturnValue({
    listen:listenFunc
})
export const mockHistory = ()=>{
    jest.resetModules();
    // jest.doMock('react-router-dom/cjs/react-router-dom', () => ({
    //     useHistory: listenFunc
    //   }));

    };
export default function testSnapFunction(oDesc, descriptions,element){
    describe(oDesc, () => {

        afterEach(() => {
            jest.clearAllMocks();
          });

        test(descriptions, () => {
            const tree = shallow(element);
            expect(tree).toMatchSnapshot();
        });
    
    })
}

