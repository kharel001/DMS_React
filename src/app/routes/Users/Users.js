import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import ListUsers from '../../../homeComponents/User/ListUsers';
import Profile from '../../../homeComponents/User/Profile/Profile';
const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    width: '100%',
    backgroundColor: '#E7E7E7'
  }

}));

const Users = () => {

  const classes = useStyles();
  

  return (
    <div>
      <Grid container className={classes.root} >
        <Switch>
          <Route path="/app/users/list-users" render={() => <ListUsers/>}/>
          <Route path="/app/users/profile" render={(props) => <Profile {...props}/>}/>
          <Redirect from="/app/users" to="/app/users/list-users" />
        </Switch>
      </Grid>

    </div>
  )
}

export default Users;
