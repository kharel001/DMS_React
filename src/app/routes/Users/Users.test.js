import testSnapFunction from '../../../util/testSnapFunction';
import Users from './Users'
import React from 'react';
import {
    configure,
    mount
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
import { Provider } from 'react-redux';
import {MemoryRouter} from 'react-router'
import configureStore from 'redux-mock-store';
import CreateUser from '../../../homeComponents/CreateUser/CreateUser';
import EditUser from '../../../homeComponents/EditUser/EditUser';
import ListUsers from '../../../homeComponents/User/ListUsers';

const mockStore = configureStore([]);


configure({
    adapter: new EnzymeAdapter
})


const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

describe('<AddMedicine />', () => {
    let wrapper;
  
    let store;
  
    beforeEach(() => {
      store = mockStore({ 
      users:["user1", "user2"]
    });
    store.dispatch=jest.fn();
    })
  
    afterEach(() => {
      jest.clearAllMocks();
    });
    it("route",()=>{try{
        wrapper = mount( 
           <Provider store={store} >
           <MemoryRouter initialEntries={['/random']}>   
          <Users users={{users1:"sa", users2:"as"}}/> 
         </MemoryRouter>  
         </Provider>
   
         );  
         expect(wrapper).toMatchSnapshot();
        }
        catch{}
   })
      
it("route",()=>{try{
     wrapper = mount( 
        <Provider store={store} >
        <MemoryRouter initialEntries={['/random']}>   
       <Users users={{users1:"sa", users2:"as"}}/> 
      </MemoryRouter>  
      </Provider>

      );  
      expect(wrapper.find(ListUsers)).toHaveLength(0);
      expect(wrapper.find(EditUser)).toHaveLength(0);
      expect(wrapper.find(CreateUser)).toHaveLength(0);
     }
     catch{}
})
it("route",()=>{
    try{

    wrapper = mount( 
        <Provider store={store} >
       <MemoryRouter initialEntries={['/app/users/list-users']}>   
      <Users users={{users1:"sa", users2:"as"}}/> 
     </MemoryRouter>        </Provider>

     );  
     expect(wrapper.find(ListUsers)).toHaveLength(1);
     expect(wrapper.find(EditUser)).toHaveLength(0);
     expect(wrapper.find(CreateUser)).toHaveLength(0);
        }
        catch{}
})
it("route",()=>{
    try{
    wrapper = mount( 
        <Provider store={store} >
       <MemoryRouter initialEntries={['/app/users/create-user']}>   
    <Users users={{users1:"sa", users2:"as"}}/> 
     </MemoryRouter>  
     </Provider>

     );  
     expect(wrapper.find(ListUsers)).toHaveLength(0);
     expect(wrapper.find(EditUser)).toHaveLength(0);
     expect(wrapper.find(CreateUser)).toHaveLength(1);
    }
    catch{}
})
it("route",()=>{
    try{
    wrapper = mount( 
        <Provider store={store} >
       <MemoryRouter initialEntries={['/app/users/edit-user']}>   
      <Users users={{users1:"sa", users2:"as"}}/> 
     </MemoryRouter>  
           </Provider>

     );  
     expect(wrapper.find(ListUsers)).toHaveLength(0);
     expect(wrapper.find(EditUser)).toHaveLength(1);
     expect(wrapper.find(CreateUser)).toHaveLength(0);
    }catch{

    }
})
})