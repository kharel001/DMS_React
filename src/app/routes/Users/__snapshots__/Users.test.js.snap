// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddMedicine /> route 1`] = `ReactWrapper {}`;
